File: mcp-server/packages/context-extractor-csharp/tests/Examples/DatabaseConnection.cs
namespace ContextExtractor.Tests.Examples;

/// <summary>
/// Manages database connections.
/// </summary>
public class DatabaseConnection
{
    public bool Connect(string host = "localhost", int port = 5432)
    {
        return true;
    }

    public void Disconnect()
    {
    }

    public List<object> ExecuteQuery(string query, List<object>? parameters = null)
    {
        return new List<object>();
    }
}

File: mcp-server/packages/context-extractor-java/src/main/java/io/contextextractor/examples/DatabaseConnection.java
package io.contextextractor.examples;

import java.util.List;
import java.util.ArrayList;

/**
 * Manages database connections.
 */
public class DatabaseConnection {
    public boolean connect(String host, int port) {
        return true;
    }
    
    public void disconnect() {
    }
    
    public List<Object> executeQuery(String query, List<Object> params) {
        return new ArrayList<>();
    }
}
